src/Orleans/Providers/ILeaseProvider.cs

```diff
@@ -27,6 +27,21 @@ public class AcquiredLease
         /// Caller side start time for this lease, which is when the lease is acquired or renewed
         /// </summary>
         public DateTime StartTimeUtc { get; }
+
+        /// <summary>
+        /// Consructor
+        /// </summary>
+        /// <param name="resourceKey"></param>
+        /// <param name="duration"></param>
+        /// <param name="token"></param>
+        /// <param name="startTimeUtc"></param>
+        public AcquiredLease(string resourceKey, TimeSpan duration, string token, DateTime startTimeUtc)
```

Comment:
This makes these classes immutable, which is good, imo.  Since they are immutable, they can be marked as immutable (attribute) which will make grain calls using them faster.